client/test-data.cpp

```diff
@@ -187,75 +157,53 @@ namespace tec {
+	}
 
 	void ProtoLoadEntity(const FilePath& fname) {
-		auto _log = spdlog::get("console_log");
-		if (fname.isValidPath() && fname.FileExists()) {
-			std::fstream input(fname.GetNativePath(), std::ios::in | std::ios::binary);
-			std::shared_ptr<EntityCreated> data = std::make_shared<EntityCreated>();
-			data->entity.ParseFromIstream(&input);
-			data->entity_id = data->entity.id();;
-			EventSystem<EntityCreated>::Get()->Emit(data);
-			for (int i = 0; i < data->entity.components_size(); ++i) {
-				const proto::Component& comp = data->entity.components(i);
-				if (in_functors.find(comp.component_case()) != in_functors.end()) {
-					in_functors[comp.component_case()](data->entity, comp);
-					entity_out_functors[data->entity_id].insert(&out_functors.at(comp.component_case()));
-				}
+		std::shared_ptr<EntityCreated> data = std::make_shared<EntityCreated>();
+		std::string json_string = LoadJSON(fname);
+		google::protobuf::util::JsonStringToMessage(json_string, &data->entity);
+		data->entity_id = data->entity.id();;
+		EventSystem<EntityCreated>::Get()->Emit(data);
+		for (int i = 0; i < data->entity.components_size(); ++i) {
+			const proto::Component& comp = data->entity.components(i);
+			if (in_functors.find(comp.component_case()) != in_functors.end()) {
+				in_functors[comp.component_case()](data->entity, comp);
 			}
 		}
-		else {
-			_log->error() << "[ProtoLoadEntity] Error opening " << fname.FileName() << " file. Can't find it";
-		}
 	}
 
 	void ProtoLoad() {
-		const std::string file("test.proto");
+		const std::string file("json/test.json");
 
 		auto _log = spdlog::get("console_log");
 		FilePath fname = FilePath::GetAssetPath(file);
 		if (fname.isValidPath() && fname.FileExists()) {
-			std::fstream input(fname.GetNativePath(), std::ios::in | std::ios::binary);
+			std::string json_string = LoadJSON(fname);
 			proto::EntityFileList elist;
-			elist.ParseFromIstream(&input);
+			google::protobuf::util::JsonStringToMessage(json_string, &elist);
 			_log->debug() << "[ProtoLoad] :\n" << elist.DebugString();
 			for (int i = 0; i < elist.entity_file_list_size(); i++) {
 				FilePath entity_filename = FilePath::GetAssetPath(elist.entity_file_list(i));
-				ProtoLoadEntity(entity_filename);
-			}
-		}
-		else {
-			_log->error() << "[ProtoLoad] Error opening " << fname.FileName() << " file. Can't find it\n";
-		}
-	}
-
-	void ProtoSave() {
-		const std::string file("test.proto");
-
-		auto _log = spdlog::get("console_log");
-		FilePath fname = FilePath::GetAssetPath("test.proto");
-		if (fname.isValidPath()) {
-			std::fstream output(fname.GetNativePath(), std::ios::out | std::ios::trunc | std::ios::binary);
-			proto::EntityFileList elist;
-			for (auto entity_functors : entity_out_functors) {
-				proto::Entity entity;
-				entity.set_id(entity_functors.first);
-				for (auto functor : entity_functors.second) {
-					(*functor)(&entity);
+				if (entity_filename.isValidPath() && entity_filename.FileExists()) {
```

Comment:
This has a time-of-check-to-time-of-use bug. If it's deleted after this check is made, it will crash. It'd be better to just always try to open, then if the file isn't there, handle that error.